torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -0,0 +1,181 @@
+  const c10::QualifiedName backend_creator_prefix(
+      backend_prefix, "__creator__");
+  const c10::QualifiedName backend_preprocess_prefix(
+      backend_prefix, "__preprocess__");
+
+  // Schemas for primitives that enable invocation of the registered create,
+  // compile and execute methods from TorchScript.
+  constexpr auto create_method_schema =
+      "prim::backend_create(str a, Any a) -> Capsule";
+  constexpr auto compile_method_schema =
+      "prim::backend_compile(Capsule c, Any a, Dict(str, Any) b) -> Dict(str, Any)";
+  constexpr auto execute_method_schema =
+      "prim::backend_execute(Capsule c, Any a, Any input) -> Any";
+
+  // Register operators that call the creator method, Backend::compile and
+  // Backend::execute.
+  torch::jit::RegisterOperators reg({
+      Operator(
+          create_method_schema,
+          [](Stack& stack) {
+            auto module = pop(stack);
+            auto qual_name = pop(stack).toStringRef();
+            auto* create_fn = getCustomClassMethod(qual_name);
+            TORCH_INTERNAL_ASSERT(create_fn);
+            push(stack, module);
+            create_fn->run(stack);
+            return 0;
+          },
+          c10::AliasAnalysisKind::FROM_SCHEMA),
+      Operator(
+          compile_method_schema,
+          [](Stack& stack) {
+            auto method_compile_spec = pop(stack).toGenericDict();
+            auto module = pop(stack);
+            auto capsule = pop(stack).toCapsule();
+            auto backend = at::dynamic_intrusive_pointer_cast<
+                PyTorchBackendInterface,
+                CustomClassHolder>(capsule);
+            TORCH_INTERNAL_ASSERT(backend);
+            auto res = backend->compile(module, method_compile_spec);
+            push(stack, res);
+            return 0;
+          },
+          c10::AliasAnalysisKind::FROM_SCHEMA),
+      Operator(
+          execute_method_schema,
+          [](Stack& stack) {
+            auto input = pop(stack);
+            auto handle = pop(stack);
+            auto capsule = pop(stack).toCapsule();
+            auto backend = at::dynamic_intrusive_pointer_cast<
+                PyTorchBackendInterface,
+                CustomClassHolder>(capsule);
+            TORCH_INTERNAL_ASSERT(backend);
+            auto res = backend->execute(handle, input);
+            push(stack, res);
+            return 0;
+          },
+          c10::AliasAnalysisKind::FROM_SCHEMA),
+  });
+
+  // For each backend:
+  for (auto& fn : getCustomClassMethodsWithPrefix(backend_creator_prefix)) {
+    auto backend_name = fn->name();
+    auto backend_creator_name = fn->qualname();
+
+    // Create a method named to_<backend> and pybind it.
+    auto m = py::handle(module).cast<py::module>();
+    std::string to_backend_method_name = "_jit_to_" + backend_name;
+    m.def(
+        to_backend_method_name.c_str(),
+        [=](Module orig_module, py::dict method_compile_spec) {
```

Comment:
We might want to store this as a `std::function` somewhere so it can be accessed from C++ as well. That might also involve separating it from pybind11 (e.g. change `py::dict`). Don't know if that should be done on this PR, though.

cc @suo